src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -43,11 +44,18 @@
 #define stringify(x)  #x
 #define expand_and_stringify(x) stringify(x)
 
-// Path from site-packages to packages that contain NuPIC Python regions
-static std::vector<const char *> packages { "nupic.regions", "nupic.regions.extra" };
-
 namespace nupic
 {
+  // Path from site-packages to packages that contain NuPIC Python regions
+  static std::vector<const char *> packages { "nupic.regions", "nupic.regions.extra" };
+
+  // Mappings for c++ regions
+  static std::map<const std::string, GenericRegisteredRegionImpl*> cpp_packages;
```

Comment:
When do the objects in this map get freed? I'd say make the value not a pointer and use `emplace` to insert new items but I don't think that works when you want to put subclasses in.